File: backend/booking-service/src/main/java/com/tripify/booking_service/dto/BookingLineDTO.java
package com.tripify.booking_service.dto;

import java.math.BigDecimal;
import java.time.LocalDate;

public record BookingLineDTO(
        Long id,
        Long catalogItemId,
        BigDecimal price,
        Integer quantity,
        Long roomTypeId,
        Long fareClassId,
        LocalDate checkIn,
        LocalDate checkOut,
        int passengerCount
) {}

File: frontend/app/src/main/java/com/tripify/tripify_android/data/model/BookingLineDTO.kt
package com.tripify.tripify_android.data.model

data class BookingLineDTO(
    val id: Long,
    val catalogItemId: Long,
    val price: Double,
    val quantity: Int? = null,
    val roomTypeId: Long? = null,
    val fareClassId: Long? = null,
    val checkIn: String? = null, // "yyyy-MM-dd"
    val checkOut: String? = null,
    val passengerCount: Int = 0
)
